crates/runtime/src/lib.rs

```diff
@@ -313,9 +374,26 @@ impl Runtime {
     }
 
     pub async fn update_dataset(&self, ds: &Dataset, all_datasets: &[Dataset]) {
```

Comment:
I really like the `test_connection`. Not sure it is the right thing to do to not unload the original dataset when updated definition can't be loaded.